clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -357,8 +290,8 @@ PidAndEpoch pidAndEpoch() {
      * Set the pid and epoch atomically. This method will signal any callers blocked on the `pidAndEpoch` method
      * once the pid is set. This method will be called on the background thread when the broker responds with the pid.
      */
-    synchronized void setPidAndEpoch(long pid, short epoch) {
-        this.pidAndEpoch = new PidAndEpoch(pid, epoch);
+    void setPidAndEpoch(PidAndEpoch pidAndEpoch) {
```

Comment:
Yes. there used to be wait/notify methods, but they were removed in a previous patch without updating the java doc.